### PR TITLE
feat(CO-53): PUT /collect (collect reward)

### PR DESCRIPTION
**Pull request by rrtoth**

### CODE REVIEW

File: test/collect-reward-test.js [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+      .put('/collect')
+      .set('Authorization', newAccTokenHeader)
+      .set('heroid', 'asd')
+      .end((err, data) => {
+        if (err) return done();
+        expect(JSON.stringify(data.body)).to.equal(JSON.stringify('Please enter a valid heroId'));
+        return done();
+      });
+  });
+});
```
Comment (Galicz555) [inline test/collect-reward-test.js]: Same problem i have in my earlier backend task that I still need to fix. You don't need a different describe test to login. You can do it in the exact describe test as a before() function

**Review by Galicz555 — COMMENTED** [review]
Pls change it

File: src/controllers/dungeonController.js [hunk 2]
```diff
@@ -15,6 +16,19 @@ class DungeonController {
...
       });
   }
+
+  collectReward(req, res) {
+    this.dungeonService.collect(req.headers.heroid)
+      .then(data => res.status(200).json(data))
+      .catch((err) => {
+        const errorResponse = {
+          400: 'Please enter a valid heroId',
+          412: 'Dungeon not finnished yet',
```
Comment (matyaspeterfi) [inline src/controllers/dungeonController.js]: The finnish are a good people, they drink a lot and like death metal, but I don't think they belong here.

**Review by matyaspeterfi — CHANGES_REQUESTED** [review]

File: src/controllers/dungeonController.js (hunk 2 above)
Comment (rrtoth) [reply src/controllers/dungeonController.js]: corrected

File: test/collect-reward-test.js (hunk 1 above)
Comment (rrtoth) [reply test/collect-reward-test.js]: corrected, BUT - I think if there is a task for it, then it should have been okay to leave it as it was

**Review by matyaspeterfi — APPROVED** [review]

**Review by juhaszPeterDev — APPROVED** [review]

**Pull request merged**